From 6e33d160ebd749d04fc967d8df07b159aa834bd6 Mon Sep 17 00:00:00 2001
From: Enric Balletbo i Serra <eballetbo@iseebcn.com>
Date: Wed, 21 Sep 2011 12:38:51 +0200
Subject: [PATCH 09/44] OMAP3: igep0020: Use common display initialitzation functions

The display initialitzation function is common for all IGEP boards, so
remove current initialitzation and use the init function provided for
the board-igep00x0.c file.

Signed-off-by: Enric Balletbo i Serra <eballetbo@iseebcn.com>
---
 arch/arm/mach-omap2/board-igep0020.c |   65 ++++++++++++++++++---------------
 1 files changed, 35 insertions(+), 30 deletions(-)

diff --git a/arch/arm/mach-omap2/board-igep0020.c b/arch/arm/mach-omap2/board-igep0020.c
index 1cfc936..63e59a4 100644
--- a/arch/arm/mach-omap2/board-igep0020.c
+++ b/arch/arm/mach-omap2/board-igep0020.c
@@ -49,7 +49,7 @@
 #define GPIO_LED_D440_GREEN		26
 #define GPIO_LED_D440_RED		27
 #define GPIO_LED_D240_RED		28
-#define IGEP2_GPIO_DVI_PUP      170
+#define GPIO_DVI_PUP			170
 
 #define IGEP2_RB_GPIO_WIFI_NPD     94
 #define IGEP2_RB_GPIO_WIFI_NRESET  95
@@ -198,48 +198,51 @@ static struct twl4030_gpio_platform_data igep0020_twl4030_gpio_pdata = {
 	.setup		= igep0020_twl_gpio_setup,
 };
 
-static int igep2_enable_dvi(struct omap_dss_device *dssdev)
+static int igep0020_enable_dvi(struct omap_dss_device *dssdev)
 {
-	gpio_direction_output(IGEP2_GPIO_DVI_PUP, 1);
+	gpio_direction_output(GPIO_DVI_PUP, 1);
 
 	return 0;
 }
 
-static void igep2_disable_dvi(struct omap_dss_device *dssdev)
+static void igep0020_disable_dvi(struct omap_dss_device *dssdev)
 {
-	gpio_direction_output(IGEP2_GPIO_DVI_PUP, 0);
+	gpio_direction_output(GPIO_DVI_PUP, 0);
 }
 
-static struct panel_generic_dpi_data dvi_panel = {
-	.name			= "generic",
-	.platform_enable	= igep2_enable_dvi,
-	.platform_disable	= igep2_disable_dvi,
+static struct omap_dss_device *dss_devices[] = {
+	&igep00x0_dvi_device,
+	&igep00x0_tv_device,
+	&igep00x0_lcd43_device,
+	&igep00x0_lcd70_device,
 };
 
-static struct omap_dss_device igep2_dvi_device = {
-	.type			= OMAP_DISPLAY_TYPE_DPI,
-	.name			= "dvi",
-	.driver_name		= "generic_dpi_panel",
-	.data			= &dvi_panel,
-	.phy.dpi.data_lines	= 24,
+static struct omap_dss_board_info dss_board_data = {
+	.num_devices	= ARRAY_SIZE(dss_devices),
+	.devices	= dss_devices,
+	.default_device	= &igep00x0_dvi_device,
 };
 
-static struct omap_dss_device *igep2_dss_devices[] = {
-	&igep2_dvi_device
-};
-
-static struct omap_dss_board_info igep2_dss_data = {
-	.num_devices	= ARRAY_SIZE(igep2_dss_devices),
-	.devices	= igep2_dss_devices,
-	.default_device	= &igep2_dvi_device,
+static struct platform_device igep0020_dss_device = {
+	.name	= "omapdss",
+	.id	= -1,
+	.dev	= {
+		.platform_data = &dss_board_data,
+	},
 };
 
-static void __init igep2_display_init(void)
+static inline void igep0020_display_init(void)
 {
-	int err = gpio_request_one(IGEP2_GPIO_DVI_PUP, GPIOF_OUT_INIT_HIGH,
-				   "GPIO_DVI_PUP");
-	if (err)
-		pr_err("IGEP v2: Could not obtain gpio GPIO_DVI_PUP\n");
+	if ((gpio_request(GPIO_DVI_PUP, "DVI PUP") == 0) &&
+	    (gpio_direction_output(GPIO_DVI_PUP, 1) == 0))
+		gpio_export(GPIO_DVI_PUP, 0);
+	else
+		pr_err("IGEP: Could not obtain gpio DVI PUP\n");
+
+	igep00x0_dvi_device.platform_enable = igep0020_enable_dvi;
+	igep00x0_dvi_device.platform_disable = igep0020_disable_dvi;
+
+	platform_device_register(&igep0020_dss_device);
 }
 
 static int igep0020_keymap[] = {
@@ -340,6 +343,10 @@ static void __init igep_init(void)
 	/* - Ethernet with SMSC9221 LAN Controller */
 	igep00x0_smsc911x_init(SMSC911X_CS, SMSC911X_IRQ, SMSC911X_NRESET);
 
+	/* Display initialitzation */
+	omap_display_init(&dss_board_data);
+	igep0020_display_init();
+
 	igep0020_leds_init();
 
 	/*
@@ -358,8 +365,6 @@ static void __init igep_init(void)
 			IGEP2_RC_GPIO_WIFI_NRESET, IGEP2_RC_GPIO_BT_NRESET,
 			!opt);
 
-	omap_display_init(&igep2_dss_data);
-	igep2_display_init();
 	usbhs_init(&igep2_usbhs_bdata);
 }
 
-- 
1.7.4.1

